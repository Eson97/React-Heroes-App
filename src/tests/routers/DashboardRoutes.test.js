import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes"

describe('Pruebas en DashboardRoutes', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name:'User'
        }
    }

    test('should show correctly', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue} >
                <MemoryRouter> {/*Simula el Router*/}
                    <DashboardRoutes />
                </MemoryRouter >
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('User');
    });


})
