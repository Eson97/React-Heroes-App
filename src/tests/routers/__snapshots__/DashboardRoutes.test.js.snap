// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DashboardRoutes should show correctly 1`] = `
Array [
  " ",
  Array [
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Heroes-App
        </a>
        <button
          aria-controls="navbarNavDropdown"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-bs-target="#navbarNavDropdown, #logout"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavDropdown"
        >
          <div
            className="navbar-nav"
          >
            <a
              aria-current="page"
              className="nav-item nav-link active"
              href="/marvel"
              onClick={[Function]}
              style={Object {}}
            >
              Marvel
            </a>
            <a
              aria-current={null}
              className="nav-item nav-link"
              href="/dc"
              onClick={[Function]}
            >
              DC
            </a>
            <a
              aria-current={null}
              className="nav-item nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </div>
        </div>
        <div
          className="navbar-collapse collapse w-100 order-3 dual-collapse2"
          id="logout"
        >
          <ul
            className="navbar-nav ms-auto"
          >
            <span
              className="nav-item nav-link text-info text-center"
            >
              User
            </span>
            <button
              className="nav-item nav-link btn"
              id="logoutBtn"
              onClick={[Function]}
            >
              Logout
            </button>
          </ul>
        </div>
      </div>
    </nav>,
    <div
      className="container mt-3"
    >
      Array [
        <h1>
          Marvel Comics
        </h1>,
        <hr />,
        <div
          className="row"
        >
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-spider"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Spider Man"
                    className="card-img"
                    src="./assets/heroes/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Spider Man
                    </h5>
                    <p
                      className="card-text"
                    >
                      Peter Parker
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Amazing Fantasy #15
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-spider"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-captain"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Captain America"
                    className="card-img"
                    src="./assets/heroes/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Captain America
                    </h5>
                    <p
                      className="card-text"
                    >
                      Steve Rogers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Captain America Comics #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-captain"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-iron"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Iron Man"
                    className="card-img"
                    src="./assets/heroes/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Iron Man
                    </h5>
                    <p
                      className="card-text"
                    >
                      Tony Stark
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #39
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-iron"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-thor"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Thor"
                    className="card-img"
                    src="./assets/heroes/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Thor
                    </h5>
                    <p
                      className="card-text"
                    >
                      Thor Odinson
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Journey into Myster #83
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-thor"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-hulk"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Hulk"
                    className="card-img"
                    src="./assets/heroes/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Hulk
                    </h5>
                    <p
                      className="card-text"
                    >
                      Bruce Banner
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hulk"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-wolverine"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Wolverine"
                    className="card-img"
                    src="./assets/heroes/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Wolverine
                    </h5>
                    <p
                      className="card-text"
                    >
                      James Howlett
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #180
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-wolverine"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-daredevil"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Daredevil"
                    className="card-img"
                    src="./assets/heroes/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Daredevil
                    </h5>
                    <p
                      className="card-text"
                    >
                      Matthew Michael Murdock
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Daredevil #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-daredevil"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-hawkeye"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Hawkeye"
                    className="card-img"
                    src="./assets/heroes/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Hawkeye
                    </h5>
                    <p
                      className="card-text"
                    >
                      Clinton Francis Barton
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #57
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hawkeye"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-cyclops"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Cyclops"
                    className="card-img"
                    src="./assets/heroes/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Cyclops
                    </h5>
                    <p
                      className="card-text"
                    >
                      Scott Summers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        X-Men #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-cyclops"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-6 col-md-4 animate__animated animate__fadeIn"
            key="marvel-silver"
          >
            <div
              className="card mt-1 mb-1"
              style={
                Object {
                  "maxWidth": 540,
                }
              }
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-md-4"
                >
                  <img
                    alt="Silver Surfer"
                    className="card-img"
                    src="./assets/heroes/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-md-8"
                >
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Silver Surfer
                    </h5>
                    <p
                      className="card-text"
                    >
                      Norrin Radd
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Fantastic Four #48
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-silver"
                      onClick={[Function]}
                    >
                      More...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>,
      ]
    </div>,
  ],
]
`;
